01-data-structures/06-trees/binary_tree/binary_search_tree.rb

```diff
@@ -3,19 +3,117 @@
 class BinarySearchTree
 
   def initialize(root)
+    @root = root
   end
 
   def insert(root, node)
+    last = nil
+    current = root
+
+    loop do 
+
+      if current.left != nil && current.rating >= node.rating 
+        current = current.left 
+      end
+
+      if current.right != nil && current.rating <= node.rating  
+        current = current.right
+      end
+
+      # if current.left == nil || current.right == nil 
+      #   if current.rating > node.rating
+      #     current.left = node
+      #     node.parent = current
+      #     break
+      #   elsif current.rating < node.rating
+      #     current.right = node
+      #     node.parent = current
+      #     break
+      #   end
+      # end
+
+      if current.left == nil 
+        if current.rating > node.rating 
```

Comment:
- [ ]  As part of the assignment, you're supposed to override `>` and `<`:

> Note that the data object must override < or > to support ordering. Hint: Use Breadth-First-Search and Depth-First-Search to implement your functions.

Not a big deal that you haven't. But for sake of illustration, you can override those functions by adding the following to `Node` if you wanted to override `>` with respect to the `rating` variable.

```ruby
 def > (otherNode)
  self.rating > otherNode.rating
 end
```